C#/avayacloud-client-subscription/ImplSubscription.cs

```diff
@@ -201,11 +217,15 @@ public async Task<List<Subscription>> getAllSubscriptions()
             Subscription createdSubscription = JObject.Parse(responseJson).ToObject<Subscription>();
             return createdSubscription;
         }
-        public async Task<bool> deleteSubscription(string subscriptionId)
+        public async Task<bool> deleteSubscription(string subscriptionId, string subAccountAppId)
         {
             await session.login();
-            SubAccount subAccount = await session.getSubAccount();
-            string subAccountAppId = subAccount.AppID;
+            //If subaccountAppId is not provided then search for one assigned.
+            if (null == subAccountAppId)
```

Comment:
this block of code is repeated multiple times, I think we can make another method called session.getSubAccountId() that returns the subAccountAppId